src/decode-stream.spec.ts

```
@@ -43,6 +43,20 @@ describe("EntityDecoder Streaming", () => {
         expect(callback).toHaveBeenCalledWith(8755, 33);
     });
 
+    it("should not over-consume a legacy compact-run entity (e.g. `&Egrave`)", () => {
+        const callback = vi.fn();
+        const decoder = new EntityDecoder(htmlDecodeTree, callback);
+
+        // `&Egrave` is a legacy (semicolon-optional) entity stored as a
+        // compact run. When it is terminated by the next character, only
+        // its 7 characters (`&Egrave`) should be consumed -- the following
+        // `&` must remain available to start the next entity.
```

ESLint

[error] Expected a block comment instead of consecutive line comments.

(multiline-comment-style)